src/model/UpsetColumn.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * Created by bikramkawan on 24/11/2016.
+ */
+import * as d3 from 'd3';
+import Column, {IColumnDesc, IStatistics} from './Column';
+import ValueColumn from './ValueColumn';
+
+
+export function numberCompare(a: number, b: number) {
+  if (isNaN(a)) { //NaN are bigger
+    return isNaN(b) ? 0 : +1;
+  }
+  if (isNaN(b)) {
+    return -1;
+  }
+  return a - b;
+}
+
+// Calculate Median, Q1 and Q1)
+export function getPercentile(data, percentile) {
+
+  var index = (percentile / 100) * data.length;
+  var result;
+  if (Math.floor(index) === index) {
+    result = (data[(index - 1)] + data[index]) / 2;
+  } else {
+    result = data[Math.floor(index)];
+  }
+  return result;
+}
+
+export class CustomSortCalculation {
+
+  constructor(private a_val: number[], private b_val: number []) {
+    this.b_val = b_val;
+    this.a_val = a_val;
+  }
+
+  countcategory() {
+
+    const a_cat = this.a_val.filter((x)=> x === 1).length;
+
+    const b_cat = this.b_val.filter((x)=> x === 1).length;
+
+    return (a_cat - b_cat);
+
+  }
+
+}
+
+export default class UpsetColumn extends ValueColumn<number[] > {
```

Comment:
since this is now an own type for representing sets, some thoughts: 
 * rename to SetColumn, Upset is just one way to render this type of column
 * is storing the information in a binary array the most efficient way?, a list/set of strings containing the sets names would be clearer
 * this would imply that this column is very similar to a categorical column. The CategoricalColumn already supports multiple values (separated by `;`).